android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/worker/MeasureReportWorker.kt

```diff
@@ -0,0 +1,150 @@
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import ca.uhn.fhir.rest.param.ParamPrefixEnum
+import com.google.android.fhir.FhirEngine
+import com.google.android.fhir.search.search
+import com.google.android.fhir.workflow.FhirOperator
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import java.util.Calendar
+import java.util.Date
+import kotlinx.coroutines.withContext
+import org.hl7.fhir.r4.model.DateTimeType
+import org.hl7.fhir.r4.model.MeasureReport
+import org.smartregister.fhircore.engine.configuration.ConfigType
+import org.smartregister.fhircore.engine.configuration.ConfigurationRegistry
+import org.smartregister.fhircore.engine.configuration.report.measure.MeasureReportConfiguration
+import org.smartregister.fhircore.engine.data.local.DefaultRepository
+import org.smartregister.fhircore.engine.util.DefaultDispatcherProvider
+import org.smartregister.fhircore.engine.util.extension.SDF_YYYY_MM_DD
+import org.smartregister.fhircore.engine.util.extension.firstDayOfMonth
+import org.smartregister.fhircore.engine.util.extension.formatDate
+import org.smartregister.fhircore.engine.util.extension.lastDayOfMonth
+import org.smartregister.fhircore.engine.util.extension.loadCqlLibraryBundle
+import org.smartregister.fhircore.engine.util.extension.parseDate
+import org.smartregister.fhircore.engine.util.extension.plusMonths
+import org.smartregister.fhircore.quest.ui.report.measure.MeasureReportViewModel
+import timber.log.Timber
+
+@HiltWorker
+class MeasureReportWorker
+@AssistedInject
+constructor(
+  @Assisted val appContext: Context,
+  @Assisted workerParams: WorkerParameters,
+  val defaultRepository: DefaultRepository,
+  val configurationRegistry: ConfigurationRegistry,
+  val dispatcherProvider: DefaultDispatcherProvider,
+  val fhirOperator: FhirOperator,
+  val fhirEngine: FhirEngine
+) : CoroutineWorker(appContext, workerParams) {
+
+  override suspend fun doWork(): Result {
+    val configuration = retrieveMeasureReportConfiguration()
+    val monthList = getMonthRangeList(configuration)
+
+    configuration.reports.forEach {
+      withContext(dispatcherProvider.io()) { fhirEngine.loadCqlLibraryBundle(fhirOperator, it.url) }
+
+      monthList.forEach { date ->
+        val startDateFormatted = date.firstDayOfMonth().formatDate(SDF_YYYY_MM_DD)
+        val endDateFormatted = date.lastDayOfMonth().formatDate(SDF_YYYY_MM_DD)
+        if (!checkReportAlreadyGenerated(startDateFormatted, endDateFormatted)) {
+          evaluatePopulationMeasure(it.url, startDateFormatted, endDateFormatted)
+        }
+      }
+    }
+
+    return Result.success()
+  }
+
+  private fun retrieveMeasureReportConfiguration(): MeasureReportConfiguration =
+    configurationRegistry.retrieveConfiguration(
+      configType = ConfigType.MeasureReport,
+      configId = "defaultMeasureReport"
+    )
```

Comment:
This private function has a single line of code and is not re-used so it is not necessary.

Reply:
This functionality needs to run for all the `MeasureReportConfiguration` available for the app. You can retrieve all configurations of the same types using `org/smartregister/fhircore/engine/configuration/ConfigurationRegistry#retrieveConfigurations` function

Reply:
The function accepts a configType and returns all the configurations matching the provided type.